src/textual_inputs/text_input.py

```diff
@@ -262,3 +279,22 @@ async def _emit_on_change(self, event: events.Key) -> None:
 
     async def _emit_on_focus(self) -> None:
         await self.emit(InputOnFocus(self))
+
+    def set_state(self, state: InputState, transient: bool = False) -> None:
```

Comment:
I like this abstraction. It could later help make `self._has_focus = True` obsolete by using the simple states and compound states like `InputState.ERROR_WITH_FOCUS` (for form validation) or similar.

Reply:
Yeah it seemed to make sense when I was writing it.. so went with it. Lets see how it turns out!